### DISCUSSION TITLE
view and navigate used libraries.

### DISCUSSION THREAD

**Discussion by dphaldes** [mfussenegger/nvim-jdtls#648] [Q&A] [answered]
I am currently checking the viability of switching to neovim for java development from IntelliJ. This project should suffice for almost of my editing needs and working with java in general.
The only prominent use that I don't see a solution for with this plugin (and other neovim plugins) is a system to look and browse through all the libraries/dependencies used by the project for examples, usage and look at implementation details. Is there something in jdtls that can facilitate this ? I am looking for any data that I can pipe into other plugins like Telescope or Trouble.nvim's tree view.
Ideally, it would be nice to have this into LSP itself and have a language agnostic method, but I don't know if such functionality even exists in the first place.

**Comment by mfussenegger** [comment, marked as answer]
There is afaik nothing like that in the language server protocol specification and I'm also not aware of any off-spec additions in eclipse.jdt.ls that would allow to retrieve the dependencies.

But would be nice to have, so I'd very likely add this if it gets implemented within eclipse.jdt.ls

**Reply by dphaldes** [reply]
It seems lsp-java for emacs has something like this integrated with treemacs https://github.com/emacs-lsp/lsp-java?tab=readme-ov-file#dependency-viewer. I will check how they do it and see if it can be replicated. Of course this is not language agnostic but from main LSP repo threads, this is not something they want to focus on ATM and think it needs to be done by the "build server protocol" instead

**Reply by dphaldes** [reply]
Apparently this has already been implemented in some form in BSP and maven extension (at least for java)
https://build-server-protocol.github.io/docs/specification/#buildtargetdependencymodules-request
https://build-server-protocol.github.io/docs/extensions/maven/#mavendependencymodule

Now the hard part is using this data and integrating it with neovim somehow. 
Regardless, I will mark this discussion as completed.

**Reply by dphaldes** [reply]
I found a plugin that does exactly this but is not maintained anymore https://github.com/NiYanhhhhh/lighttree-java
